honeybee_energy/construction/opaque.py

```diff
@@ -200,6 +201,8 @@ def from_dict(cls, data):
         new_obj = cls(data['identifier'], mat_layers)
         if 'display_name' in data and data['display_name'] is not None:
             new_obj.display_name = data['display_name']
+        if 'user_data' in data and data['user_data'] is not None:
+            opaqueConstruction.user_data = data['user_data']
```

Comment:
You also need to add a couple of lines like this into the `from_dict_abridged` classmethod.